pkg/ciliumenvoyconfig/cec_reconciler_test.go

```diff
@@ -0,0 +1,673 @@
+		},
+		{
+			name: "Add configs that start to select the local node",
+			configs: map[resource.Key]*config{
+				{Namespace: "ns1", Name: "config1"}: testConfig("ns1", "config1", map[string]string{"role": "infra"}, false),
+				{Namespace: "ns1", Name: "config2"}: testConfig("ns1", "config2", nil, true),
+			},
+			currentNodeLabels: map[string]string{
+				"role": "worker",
+			},
+			newNodeLabels: map[string]string{
+				"role": "infra",
+			},
+			expectedError:   false,
+			expectedAdded:   []string{"ns1/config1"},
+			expectedDeleted: []string{},
+		},
+		{
+			name: "Multiple changes",
+			configs: map[resource.Key]*config{
+				{Namespace: "ns1", Name: "config1"}: testConfig("ns1", "config1", map[string]string{"role": "infra", "node": "node1"}, false),
+				{Namespace: "ns1", Name: "config2"}: testConfig("ns1", "config2", map[string]string{"role": "infra", "node": "node1"}, false),
+				{Namespace: "ns1", Name: "config3"}: testConfig("ns1", "config3", map[string]string{"role": "infra", "node": "node1", "environment": "test"}, false),
+				{Namespace: "ns1", Name: "config4"}: testConfig("ns1", "config4", nil, true),
+				{Namespace: "ns1", Name: "config5"}: testConfig("ns1", "config5", map[string]string{"role": "worker", "node": "node1"}, true),
+				{Namespace: "ns1", Name: "config6"}: testConfig("ns1", "config6", map[string]string{"role": "worker", "node": "node1"}, true),
+				{Namespace: "ns1", Name: "config7"}: testConfig("ns1", "config7", map[string]string{"role": "worker", "node": "node1", "environment": "test"}, true),
+			},
+			currentNodeLabels: map[string]string{
+				"node": "node1",
+				"role": "worker",
+			},
+			newNodeLabels: map[string]string{
+				"node": "node1",
+				"role": "infra",
+			},
+			expectedError:   false,
+			expectedAdded:   []string{"ns1/config1", "ns1/config2"},
+			expectedDeleted: []string{"ns1/config5", "ns1/config6", "ns1/config7"},
+		},
+		{
+			name: "Failures during updating individual configs should't abort",
+			configs: map[resource.Key]*config{
+				{Namespace: "ns1", Name: "config1"}: testConfig("ns1", "config1", map[string]string{"role": "infra", "node": "node1"}, false),
+				{Namespace: "ns1", Name: "config2"}: testConfig("ns1", "config2", map[string]string{"role": "infra", "node": "node1"}, false),
+				{Namespace: "ns1", Name: "config3"}: testConfig("ns1", "config3", map[string]string{"role": "infra", "node": "node1", "environment": "test"}, false),
+				{Namespace: "ns1", Name: "config4"}: testConfig("ns1", "config4", nil, true),
+				{Namespace: "ns1", Name: "config5"}: testConfig("ns1", "config5", map[string]string{"role": "worker", "node": "node1"}, true),
+				{Namespace: "ns1", Name: "config6"}: testConfig("ns1", "config6", map[string]string{"role": "worker", "node": "node1"}, true),
+				{Namespace: "ns1", Name: "config7"}: testConfig("ns1", "config7", map[string]string{"role": "worker", "node": "node1", "environment": "test"}, true),
```

Comment:
same, current labels do not have `environment`: `test`, so maybe this should be `false` for consistency?

Reply:
see above - changed